doc/development/guide/tests.rst

```diff
@@ -16,26 +15,155 @@ These requirements can be installed via ``pip``:
 
     pip install pytest pytest-cov pytest-mock flaky
 
+Creating a test
+~~~~~~~~~~~~~~~
+Every test has to be added to the PennyLane `test folder <https://github.com/PennyLaneAI/pennylane/tree/master/tests>`__.
+The test folder follows the structure of the PennyLane module folder. Therefore, tests needs to be added to the corresponding subfolder of the functionality they are testing.
+
+Most tests typcally will not require the use of an interface (such as Autograd, Torch, TensorFlow and Jax). Tests without an interface will be marked
+as a ``core`` test automatically by pytest (the functionality for this is located in ``conftest.py``). For such general tests, you can follow the structure of the example below,
+where it is recommended that you follow general pytest guidelines:
```

Comment:
Maybe good to turn this into a link to the pytest documentation?

Reply:
Yes added!